pyhf/utils.py

```diff
@@ -157,19 +164,50 @@ def runOnePoint(muTest, data, pdf, init_pars=None, par_bounds=None):
     asimov_mu = 0.0
     asimov_data = generate_asimov_data(asimov_mu, data, pdf, init_pars, par_bounds)
 
-    qmu_v = tensorlib.clip(qmu(muTest, data, pdf, init_pars, par_bounds), 0, max=None)
+    qmu_v = tensorlib.clip(qmu(mu_test, data, pdf, init_pars, par_bounds), 0, max=None)
     sqrtqmu_v = tensorlib.sqrt(qmu_v)
 
     qmuA_v = tensorlib.clip(
-        qmu(muTest, asimov_data, pdf, init_pars, par_bounds), 0, max=None
+        qmu(mu_test, asimov_data, pdf, init_pars, par_bounds), 0, max=None
     )
     sqrtqmuA_v = tensorlib.sqrt(qmuA_v)
 
     CLsb, CLb, CLs = pvals_from_teststat(sqrtqmu_v, sqrtqmuA_v)
-
-    CLs_exp = []
-    for nsigma in [-2, -1, 0, 1, 2]:
-        sqrtqmu_v_sigma = sqrtqmuA_v - nsigma
-        CLs_exp.append(pvals_from_teststat(sqrtqmu_v_sigma, sqrtqmuA_v)[-1])
-    CLs_exp = tensorlib.astensor(CLs_exp)
-    return qmu_v, qmuA_v, CLsb, CLb, CLs, CLs_exp
+    _returns = CLs
+    if kwargs:
```

Comment:
I don't think this is needed.. if no kwargs are passed it should already be a empty dict

Reply:
There might be a better way of doing this that will become apparent once I've had coffee, but as if there is `kwargs` then

```
_returns = tuple(_returns)
```

so in order to try to have a consistent return signature for the CLs_obs without `kwargs` (a Tensor and not a `tuple`) I added the default and the check for the empty dict.

Reply:
Yes, of course. Sorry, I don't think I'm understanding what you're trying to show me. :? I don't see how that applies to what I'm saying RE: keeping the return signature of the CLs_obs the same regardless of `kwargs`.

```python
# Current: with guard
>>> import pyhf
>>> import pyhf.simplemodels
>>> import pyhf.utils
>>> 
>>> pdf = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
>>> CLs_obs = pyhf.utils.hypotest(1.0, [51, 48] + pdf.config.auxdata, pdf)
>>> print(CLs_obs, type(CLs_obs))
[0.05290116] <class 'numpy.ndarray'>
>>> print('Observed: {}'.format(CLs_obs))
Observed: [0.05290116]
>>> 
>>> kwargs = {'return_expected': True}
>>> CLs_obs, CLs_exp = pyhf.utils.hypotest(1.0, [51, 48] + pdf.config.auxdata, pdf, **kwargs)
>>> print('Observed: {}, Expected: {}'.format(CLs_obs, CLs_exp))
Observed: [0.05290116], Expected: [0.06445521]
```

```python
# No guard: return signature is not consistent
>>> import pyhf
>>> import pyhf.simplemodels
>>> import pyhf.utils
>>> 
>>> pdf = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
>>> CLs_obs = pyhf.utils.hypotest(1.0, [51, 48] + pdf.config.auxdata, pdf)
>>> print(CLs_obs, type(CLs_obs))
(array([0.05290116]),) <class 'tuple'>
>>> print('Observed: {}'.format(CLs_obs))
Observed: (array([0.05290116]),)
>>> 
>>> kwargs = {'return_expected': True}
>>> CLs_obs, CLs_exp = pyhf.utils.hypotest(1.0, [51, 48] + pdf.config.auxdata, pdf, **kwargs)
>>> print('Observed: {}, Expected: {}'.format(CLs_obs, CLs_exp))
Observed: [0.05290116], Expected: [0.06445521]
```

Reply:
ah, sorry I misunderstood... so if I'm reading this right you only want to return a tuple in case there are two return values, other wise just the single object, or am I issing something?

we might just check at the very and the length of `_returns` and do `_returns if len(_returns) > 1 else _returns[0]`

Reply:
Yeah, I just want to have the returned `CLs_obs` be the same either way. Without some check then in the absence of `kwargs`

```python
CLs_obs = pyhf.utils.hypotest(...)
```
would have `CLs_obs` be of type `tuple`, where 

```python
CLs_obs, some_other_return = pyhf.utils.hypotest(..., **kwargs)
```

has `CLs_obs` being assigned an element of the returned tuple instead. The returned object is obviously the same, but the difference in what type of object the variable is assigned is not desirable.

Your suggested check of

```python
if len(_returns) > 1:
    _returns = tuple(_returns)
else:
    _returns = _returns[0]
```
is a fine alternative to checking for a non-empty dict. :+1:

Reply:
ok .. sorry for the confusion!

Reply:
No no, I think cleaner code would have avoided it in the first place. :)

Reply:
out of curiosity if `return A if b else C` against black?

Reply:
It is not! Changed in https://github.com/diana-hep/pyhf/pull/341/commits/4a8225ebf4adc31ad092146a8f2b077d9fc0b781